File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task1/Java/MatrixThread.java
package lab8.threading;

public class MatrixThread implements Runnable {
    private Thread thread;
    private int[][] matrix;
    private int startRow;

    public MatrixThread(
            int[][] matr,
            String name,
            int row,
            int priority) { 
        
        matrix = matr;
        startRow = row;
        
        thread = new Thread(this);
        thread.setName(name);
        thread.setPriority(priority);
        thread.start();
    }
    
    public void run() {
        for (int i = startRow; i < matrix.length; i += 2)
            MatrFun.printRow(i, matrix);
    }
}

File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task1/C#/MatrixThread.cs
using System;
using System.Threading;

namespace ThreadingTask {

    class MatrixThread {
        // Private class members.
        private Thread thread;
        private int[,] matrix;
        private int startRow;

        public MatrixThread(
            int[,] matr,
            string name,
            int row,
            ThreadPriority priority) {

                matrix = matr;
                startRow = row;

                thread = new Thread(this.Run);
                thread.Name = name;
                thread.Priority = priority;
                thread.Start();
        }

        private void Run() {
            // Calling synchronized static method.
            for (int i = startRow; i < matrix.GetLength(0); i += 2)
                MatrFun.PrintRow(i, matrix);
        }
    }
}
